design/ceph/object/store.md

```diff
@@ -119,3 +119,48 @@ Specifying this section also ensures that the pool section in the ceph-object-zo
   zone:
     name: "name"
 ```
+
+### LDAP
+
+The Ceph Object Gateway supports integrating with LDAP for authenticating and creating users, please refer [here](https://docs.ceph.com/docs/master/radosgw/ldap-auth/). This means that the `rgw backend user` is also required to be part of groups in the LDAP server otherwise, authentication will fail. The `rgw backend user` can be generated from `CephObjectStoreUser` or `ObjectBucketClaim` CRDs. For the both resources credentials are saved in Kubernetes Secrets which may not be valid with `LDAP Server`, user need to follow the steps mentioned [here](https://docs.ceph.com/en/latest/radosgw/ldap-auth/#generating-an-access-token-for-ldap-authentication).The following settings need to be configured in the RGW server:
+```
+rgw ldap binddn =
+rgw ldap secret = /etc/ceph/ldap/bindpass.secret
+rgw ldap uri =
+rgw ldap searchdn =
+rgw ldap dnattr =
+rgw ldap searchfilter =
+rgw s3 auth use ldap = true
+```
+So the CRD for the Ceph Object Store will be modified to include the above changes:
+
+```yaml
+spec:
+  security
+    ldap:
+      config:
+        uri: ldaps://ldap-server:636
+        binddn: "uid=ceph,cn=users,cn=accounts,dc=example,dc=com"
+        searchdn: "cn=users,cn=accounts,dc=example,dc=com"
+        dnattr: "uid"
+        searchfilter: "memberof=cn=s3,cn=groups,cn=accounts,dc=example,dc=com"
+      credential:
+        secret: object-my-store-ldap-creds
```

Comment:
For `credential`, let's design this to use the same mounting scheme that is working well for us in the CephNFS design that allows users to set the credential secret from whatever source is best for them.

See the `kerberos` section here (keytab is most similar to a cert):
https://github.com/rook/rook/blob/master/design/ceph/ceph-nfs-ganesha.md#example

See how this is documented here: https://rook.io/docs/rook/v1.11/CRDs/ceph-nfs-crd/#security
And here: https://rook.io/docs/rook/v1.11/Storage-Configuration/NFS/nfs-security/#kerberos-configuration

Users may omit the credential if they wish to inject it via vault or built it into their container statically. They may mount it from a K8s Secret, or from a directory on the host if that is where the LDAP creds are stored, as well as any other volume. 

This approach is highly flexible and means that we don't need to do extra work to add support for KMSes at a later date.